hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -104,10 +98,9 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
     this.props = props;
     this.start = start;
     this.length = length;
-    this.recordMergeMode = getRecordMergeMode(props);
     HoodieTableConfig tableConfig = hoodieTableMetaClient.getTableConfig();
-    this.recordMerger = readerContext.getRecordMerger(tableConfig.getRecordMergerStrategy());
-    readerContext.setRecordMerger(this.recordMerger);
+    readerContext.setRecordMerger(readerContext.getRecordMerger(tableConfig.getRecordMergeMode(),
+        tableConfig.getRecordMergerStrategy(), props.getString("hoodie.datasource.write.record.merger.impl","")));
```

Comment:
Reference the config property declaration of "hoodie.datasource.write.record.merger.impl".

Reply:
Could we directly do `readerContext.setRecordMerger(tableConfig.getRecordMergeMode(),
        tableConfig.getRecordMergerStrategy(), props.getString("hoodie.datasource.write.record.merger.impl",""))` by changing the setter logic?

Reply:
It's good that all immutable merge configs come from table config now.  Should we persist `hoodie.datasource.write.record.merger.impl` in table config too and make it mutable, so it also comes from the table config?

Reply:
Let's define the config name in final strings so we can clean them up in the future if needed.

Reply:
HUDI-8170 for improving the reader context APIs.  The config references are fixed.